src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -556,6 +562,11 @@ internal void UpdateThemeBindings(ResourceUpdateReason updateReason)
 			{
 				mergedDict.UpdateThemeBindings(updateReason);
 			}
+
+			if (_sourceDictionary?.TryGetTarget(out var target) == true)
```

Comment:
Typed WeakReference uses generics which is slower than using a cast (for now at least). When changing to ManagedWeakReference:

```suggestion
			if (_sourceDictionary.Target is ResourceDictionary target)
```